ui/src/app/job-list/table/table.component.ts

```diff
@@ -200,4 +200,19 @@ export class JobsTableComponent implements OnInit {
       this.jobs.forEach(j => this.selection.select(j));
     }
   }
+
+  updateCheckBoxSelection(clickedJob: QueryJobsResult, event: MouseEvent): void {
+    // if the user has shift-clicked on a job, find the last selected job and
+    // select all jobs between the two
+    if (event.shiftKey && this.selection.selected) {
+      const lastJobClicked = this.selection.selected[this.selection.selected.length - 1];
+      const [from, to] = [
+        this.jobs.findIndex(j => j.id === lastJobClicked.id),
```

Comment:
nit: this is probably more accurately the `lastJobSelected`. We may have clicked another one to deselect it.